services/wallet/service.go

```diff
@@ -59,7 +60,11 @@ func NewService(
 		Publisher: walletFeed,
 	}
 	blockchainStatus := make(map[uint64]string)
+	mutex := sync.Mutex{}
```

Comment:
hmm having doubts about instancing the mutex as a local variable and using it inside the callback. Are we sure this works like we expect it to work?
Also seems like blockchainStatus works like some kind of implicit member variable, perhaps we should explicitly define this and the mutex as members for clarity